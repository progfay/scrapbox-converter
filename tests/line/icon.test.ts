/* global describe it expect */

import { parse } from '@progfay/scrapbox-parser'
import convert from '../../src'
import pretty from 'pretty'

describe('icon', () => {
  it('Simple icon', () => {
    const input = `Icon
[progfay.icon]`
    const html = convert(parse(input), 'projectName')
    expect(pretty(html)).toMatchSnapshot()
  })

  it('External icon', () => {
    const input = `Icon
[/icons/hr.icon]`
    const html = convert(parse(input), 'projectName')
    expect(pretty(html)).toMatchSnapshot()
  })
})
