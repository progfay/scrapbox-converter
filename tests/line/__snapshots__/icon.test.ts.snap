// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon External icon 1`] = `
"<article>
  <h1 class=\\"title\\">Icon</h1>
  <section>
    <p>
      <amp-img src=\\"https://scrapbox.io/api/pages/icons/hr/icon\\" layout=\\"fill\\" alt=\\"icon\\" class=\\"normal-icon\\"></amp-img>
    </p>
  </section>
</article>"
`;

exports[`icon Simple icon 1`] = `
"<article>
  <h1 class=\\"title\\">Icon</h1>
  <section>
    <p>
      <amp-img src=\\"https://scrapbox.io/api/pages/projectName/progfay/icon\\" layout=\\"fill\\" alt=\\"icon\\" class=\\"normal-icon\\"></amp-img>
    </p>
  </section>
</article>"
`;
